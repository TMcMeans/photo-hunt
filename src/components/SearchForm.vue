<template>
  <div class="search-form">
    <p v-show="error">{{error}}</p>
    <input type="text" v-model="query" placeholder="Search for photos">
    <button @click.prevent="onClick">Search</button>
  </div>
</template>

<script>
export default {
  name: "SearchForm",
  props: {},
  data: () => ({
    query: "",
    error: ""
  }),
  methods: {
    onClick() {
      if (this.query) {
        this.error = "";
        this.$emit("handleClick", this.query);
        this.query = "";
      } else {
        this.error = "Type a query in the search bar to view related images.";
      }
    }
  }
};
</script>
 
<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
input[type="text"] {
  width: 200px;
  height: 25px;
  font-size: 1em;
  margin: 10px;
  border-top: none;
  border-right: none;
  border-left: none;
  border-bottom: 2px solid black;
}
button {
  border-radius: 0%;
  height: 30px;
  font-size: 1em;
  border: 2px solid black;
  cursor: pointer;
}
button:hover {
  background: black;
  color: white;
}
</style>
