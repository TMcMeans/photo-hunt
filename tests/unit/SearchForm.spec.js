import { mount, shallowMount } from '@vue/test-utils';
import SearchForm from '@/components/SearchForm';

describe('SearchForm', () => {
  it('should render a snapshot with no data passed in', () => {
    const wrapper = mount(SearchForm);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render a snapshot with correct query data passed in', () => {
    const wrapper = mount(SearchForm);
    wrapper.setData({ query: 'trees' });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render a snapshot with correct error data passed in', () => {
    const wrapper = mount(SearchForm);

    wrapper.setData({
      error: 'Type a query in the search bar to view related images.'
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should call onClick method when button is pressed', () => {
    const wrapper = mount(SearchForm);
    wrapper.vm.onClick = jest.fn();

    wrapper.find('button').trigger('click');
    expect(wrapper.vm.onClick).toHaveBeenCalled();
  });

  it('should set error to a string if no search query is entered', () => {
    const wrapper = mount(SearchForm);

    wrapper.find('button').trigger('click');
    wrapper.vm.onClick();
    expect(wrapper.attributes().error).toEqual();
  });

  it('should set query to a string if entered in a search', () => {
    const wrapper = mount(SearchForm);

    const input = wrapper.find('input');
    input.element.value = 'kittens';
    input.trigger('click');
    expect(wrapper.attributes().query).toEqual();
  });
});
