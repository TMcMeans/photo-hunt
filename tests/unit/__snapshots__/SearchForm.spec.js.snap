// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should render a snapshot with correct error data passed in 1`] = `
<div
  class="search-form"
>
  <p
    style=""
  >
    Type a query in the search bar to view related images.
  </p>
   
  <input
    placeholder="Search for photos"
    type="text"
  />
   
  <button>
    Search
  </button>
</div>
`;

exports[`SearchForm should render a snapshot with correct query data passed in 1`] = `
<div
  class="search-form"
>
  <p
    style="display: none;"
  >
    
  </p>
   
  <input
    placeholder="Search for photos"
    type="text"
  />
   
  <button>
    Search
  </button>
</div>
`;

exports[`SearchForm should render a snapshot with no data passed in 1`] = `
<div
  class="search-form"
>
  <p
    style="display: none;"
  >
    
  </p>
   
  <input
    placeholder="Search for photos"
    type="text"
  />
   
  <button>
    Search
  </button>
</div>
`;
